
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../..">
      
      
        <link rel="next" href="../assembly_free/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Assembly based - Metagenomics</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-purple" data-md-color-accent="deep-orange">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#metagenome-assembly-and-binning" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Metagenomics" class="md-header__button md-logo" aria-label="Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Metagenomics
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Assembly based
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
  
    
  
  Home

      </a>
    </li>
  

      
        
  
  
  
    
  
  
    <li class="md-tabs__item md-tabs__item--active">
      <a href="./" class="md-tabs__link">
        
  
  
    
  
  Assembly based

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../assembly_free/" class="md-tabs__link">
        
  
  
    
  
  Assembly free

      </a>
    </li>
  

      
        
  
  
  
  
    <li class="md-tabs__item">
      <a href="../metabarcoding/" class="md-tabs__link">
        
  
  
    
  
  Metabarcoding

      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Metagenomics" class="md-nav__button md-logo" aria-label="Metagenomics" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    Metagenomics
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Home
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Assembly based
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Assembly based
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Getting the Data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      Quality Control
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#binning" class="md-nav__link">
    <span class="md-ellipsis">
      Binning
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#checking-the-quality-of-the-bins" class="md-nav__link">
    <span class="md-ellipsis">
      Checking the quality of the bins
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#functional-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Functional annotation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomic annotation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      Further reading
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../assembly_free/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assembly free
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../metabarcoding/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Metabarcoding
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-data" class="md-nav__link">
    <span class="md-ellipsis">
      Getting the Data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quality-control" class="md-nav__link">
    <span class="md-ellipsis">
      Quality Control
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#assembly" class="md-nav__link">
    <span class="md-ellipsis">
      Assembly
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#binning" class="md-nav__link">
    <span class="md-ellipsis">
      Binning
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#checking-the-quality-of-the-bins" class="md-nav__link">
    <span class="md-ellipsis">
      Checking the quality of the bins
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#functional-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Functional annotation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-annotation" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomic annotation
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#further-reading" class="md-nav__link">
    <span class="md-ellipsis">
      Further reading
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  



<h1 id="metagenome-assembly-and-binning">Metagenome assembly and binning</h1>
<p>Before anything RUN THIS:</p>
<pre><code class="language-bash">conda create -n Meta_assembly -c bioconda fastqc sickle-trim megahit bowtie2 samtools metabat2=2.15 checkm-genome prokka
conda create -n sour -c bioconda sourmash=4.5.0
</code></pre>
<p>In this tutorial you'll learn how to inspect assemble metagenomic data and retrieve draft genomes from assembled metagenomes</p>
<p>We'll use a mock community of 20 bacteria sequenced using the Illumina HiSeq.
In reality the data were simulated using <a href="http://insilicoseq.readthedocs.io">InSilicoSeq</a>.</p>
<p>The 20 bacteria in the dataset were selected from the <a href="http://ocean-microbiome.embl.de/companion.html">Tara Ocean study</a> that recovered 957 distinct Metagenome-assembled-genomes (or MAGs) that were previsouly unknown! (full list on <a href="https://figshare.com/articles/TARA-NON-REDUNDANT-MAGs/4902923/1">figshare</a> )</p>
<h2 id="getting-the-data">Getting the Data</h2>
<pre><code class="language-bash">mkdir -p ~/data
cd ~/data
curl -O -J -L https://osf.io/th9z6/download
curl -O -J -L https://osf.io/k6vme/download
chmod -w tara_reads_R*
</code></pre>
<h2 id="quality-control">Quality Control</h2>
<p>we'll use FastQC to check the quality of our data, as well as sickle for trimming the bad quality part of the reads.
If you need a refresher on how and why to check the quality of sequence data, please check the <a href="qc">Quality Control and Trimming</a> tutorial</p>
<pre><code class="language-bash">mkdir -p ~/results
cd ~/results
ln -s ~/data/tara_reads_* .
fastqc tara_reads_*.fastq.gz
</code></pre>
<p>!!! question
    What is the average read length? The average quality?</p>
<p>!!! question
    Compared to single genome sequencing, what graphs differ?</p>
<p>Now we'll trim the reads using sickle</p>
<pre><code>sickle pe -f tara_reads_R1.fastq.gz -r tara_reads_R2.fastq.gz -t sanger \
    -o tara_trimmed_R1.fastq -p tara_trimmed_R2.fastq -s /dev/null
</code></pre>
<p>!!! question
    How many reads were trimmed?</p>
<h2 id="assembly">Assembly</h2>
<p>Megahit will be used for the assembly.</p>
<pre><code>megahit -1 tara_trimmed_R1.fastq -2 tara_trimmed_R2.fastq -o tara_assembly
</code></pre>
<p>the resulting assembly can be found under <code>tara_assembly/final.contigs.fa</code>.</p>
<p>!!! question
    How many contigs does this assembly contain?</p>
<h2 id="binning">Binning</h2>
<p>First we need to map the reads back against the assembly to get coverage information</p>
<pre><code class="language-bash">ln -s tara_assembly/final.contigs.fa .
bowtie2-build final.contigs.fa final.contigs
bowtie2 -x final.contigs -1 tara_reads_R1.fastq.gz -2 tara_reads_R2.fastq.gz | \
    samtools view -bS -o tara_to_sort.bam
samtools sort tara_to_sort.bam -o tara.bam
samtools index tara.bam
</code></pre>
<p>then we run metabat</p>
<pre><code class="language-bash">runMetaBat.sh -m 1500 final.contigs.fa tara.bam
mv final.contigs.fa.metabat-bins1500 metabat
</code></pre>
<p>!!! question
    How many bins did we obtain?</p>
<h2 id="checking-the-quality-of-the-bins">Checking the quality of the bins</h2>
<p>The first time you run <code>checkm</code> you have to create the database</p>
<pre><code class="language-bash">mkdir   db/
wget --no-check-certificate https://data.ace.uq.edu.au/public/CheckM_databases/checkm_data_2015_01_16.tar.gz
mv  checkm_data_2015_01_16.tar.gz db/
cd db/
tar -xzvf checkm_data_2015_01_16.tar.gz
cd ..
checkm data setRoot db/
</code></pre>
<pre><code class="language-bash">checkm lineage_wf -x fa metabat checkm/
checkm qa  checkm/lineage.ms checkm &gt; checkm.txt
</code></pre>
<p>!!! question
    Which bins should we keep for downstream analysis?</p>
<h2 id="functional-annotation">Functional annotation</h2>
<p>Now we should have a collection of MAGs that we can further analyze. The first step is to predict genes as right now we only have raw genomic sequences. We will use one of my all-time-favorites : prokka.</p>
<p>This tool does gene prediction as well as some decent and useful annotations, and is actually quite easy to run!</p>
<p>Prokka produces a number of output files that all kinds of represent similar things. Mostly variants of FASTA-files, one with the genome again, one with the predicted proteins, one with the genes of the predicted proteins. Also, it renames all the sequences with nicer IDs! Additionally, a very useful file generated is a GFF-file, which gives more information about the annotations than just the names you can see in the FASTA-files.</p>
<p>The annotations of prokka are good but not very complete for environmental bacteria. Let’s run an other tool I like a lot, eggNOGmapper. This is a bit heavier in computation.</p>
<pre><code>Install eggNOG-mapper run it on at least one MAG (don’t be greedy, it is not fast!) OPTIONAL : undestand the output of it …
</code></pre>
<h2 id="taxonomic-annotation">Taxonomic annotation</h2>
<p>We now know more about the genes your MAG contains, however we do not really know who we have?! checkm might have given us an indication but it is only approximative.</p>
<p>Taxonomic classification for full genomes is not always easy for MAGs, often the 16S gene is missing as it assembles badly, and which other genes to use to for taxonomy is not always evident. Typically marker genes, min-hashes or k-mer databases are used as reference. It is often problematic for environmental data as the databases are not biased into our direction! We will use a min-hash database I compiled specifically for this (based on other tools and the full-datasets) using a tool called <a href="https://sourmash.readthedocs.io/en/latest/">sourmash</a></p>
<pre><code>Install sourmash in your computer!

```bash
conda deactivate
conda activate sour
sourmash compute -k 31 --scaled 10000 metabat/bin.* 
```

to compute signatures Use the lca clasify function of sourmash with the database you can find here

```bash
wget https://osf.io/download/p9ezm -O gtdb-rs207.genomic-reps.dna.k31.lca.json.gz

sourmash lca classify --query bin.1.fa.sig --db gtdb-rs207.genomic-reps.dna.k31.lca.json.gz
```
</code></pre>
<p>This database was made by the representative genomes of the gtdb database</p>
<p>Links to an external site.. The GTDB database is also used by the tool gtdbtk, it uses marker genes and loads of data. It is a bit heavy, and tricky to run/install but much more sensitive.</p>
<pre><code>Optional: install and run gtdbtk
</code></pre>
<h2 id="further-reading">Further reading</h2>
<ul>
<li><a href="https://www.nature.com/articles/s41564-017-0012-7">Recovery of nearly 8,000 metagenome-assembled genomes substantially expands the tree of life</a></li>
<li><a href="https://www.nature.com/articles/sdata2017203">The reconstruction of 2,631 draft metagenome-assembled genomes from the global oceans</a></li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tabs"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>